clang/test/dpct/syncthreads_in_cond.cu

```diff
@@ -0,0 +1,106 @@
+// UNSUPPORTED: cuda-8.0
+// UNSUPPORTED: v8.0
+// RUN: dpct --format-range=none -out-root %T/syncthreads_in_cond %s --cuda-include-path="%cuda-path/include" --use-experimental-features=nd_range_barrier,logical-group -- -x cuda --cuda-host-only -std=c++14
+// RUN: FileCheck %s --input-file %T/syncthreads_in_cond/syncthreads_in_cond.dp.cpp
+
+#include <cuda_runtime.h>
+
+__device__ void test1(uint64_t *a) {
+  unsigned tid =
+      ((blockIdx.x + (blockIdx.y * gridDim.x)) * (blockDim.x * blockDim.y)) +
+      (threadIdx.x + (threadIdx.y * blockDim.x));
+
+  // check-achor-begin-1
+  __syncthreads();
+  // check-achor-end-1
+  // CHECK: check-achor-begin-1
+  // CHECK-NOT: DPCT1112:{{.*}}: Synchronizing thread blocks inside a conditional branch statement may lead to undefined behavior
+  // CHECK: check-achor-end-1
+
+  // switch
+  switch (tid) {
+  case 0:
+    a[0] = 1;
+    // CHECK: DPCT1112:{{.*}}: Synchronizing thread blocks inside a conditional branch statement may lead to undefined behavior
+    __syncthreads();
+    break;
+  case 1:
+    a[1] = 1;
+    // CHECK: DPCT1112:{{.*}}: Synchronizing thread blocks inside a conditional branch statement may lead to undefined behavior
+    __syncthreads();
+    break;
+  default:
+    a[tid] = a[tid - 1] + a[tid - 2];
+  }
+
+  // if
+  if (tid > 32) {
+    // CHECK: DPCT1112:{{.*}}: Synchronizing thread blocks inside a conditional branch statement may lead to undefined behavior
+    __syncthreads();
+  } else {
+    a[tid] = 0;
+    // CHECK: DPCT1112:{{.*}}: Synchronizing thread blocks inside a conditional branch statement may lead to undefined behavior
+    __syncthreads();
+  }
+
+  // do
+  do {
+    a[tid] = a[tid - 1] + a[tid - 2];
+    // CHECK: DPCT1112:{{.*}}: Synchronizing thread blocks inside a conditional branch statement may lead to undefined behavior
+    __syncthreads();
+  } while (a[tid]);
+
+  // while
+  while (a[tid])
+    // CHECK: DPCT1112:{{.*}}: Synchronizing thread blocks inside a conditional branch statement may lead to undefined behavior
+    __syncthreads();
+
+  // early return
+  if (tid < 32)
+    return;
+
+  // CHECK: DPCT1112:{{.*}}: Synchronizing thread blocks inside a conditional branch statement may lead to undefined behavior
```

Comment:
let's not emit Line62. instead emit in the callee of test1.

Reply:
done